Testscripts/Linux/channel_change.sh

```diff
@@ -135,59 +143,137 @@ function Main() {
-			LogErr "Found the currect cpu $id is not online. Expected 1 but found $state"
+			LogErr "Found the currect cpu $id was not online. Expected 1 but found $state"
 			FailedCount=$((FailedCount+1))
 		fi
 	done
 
+	# ########################################################################
+	# Assign a random CPU id to the vmbus channel
+	# Assign the original CPU id to the vmbus channel
+	# Now each vmbus's cpu id is 0.
+
+	# Find the max CPU core counts
+	max_cpu=$(nproc)
+
+	LogMsg "Re-read the original lsvmbus output file"
+	while IFS=' ' read -a line
+	do
+		if [[ ${line[0]} == "VMBUS" && ${line[1]} == "ID" ]]; then
+			vmbus_id[$idx]=${line[2]%?}
+			LogMsg "Found new VMBUS ID --> ${vmbus_id[$idx]}"
+
+			read -a line
+			# Ignored Device Id
+
+			read -a line
+			if [[ ${line[0]} =~ "Sysfs" && ${line[1]} == "path:" ]]; then
+				sysfs_path=${line[2]}
+				LogMsg "Found new sysfs --> $sysfs_path"
+
+				# Read Rel_ID and target_cpu lines
+				read -a line
+				while [[ $line != '' ]]; do
+					rel_id=${line[0]#"$prefix1"}
+					rel_id=${rel_id%"$suffix1"}
+					LogMsg "Found Rel_ID: $rel_id, which is vmbus channel id"
+					cpu_id=${line[1]#"$prefix2"}
+					LogMsg "Found the original target cpu id: $cpu_id"
+					# This cpu_is is from lsvmbus.output, the original output.
+					if [[ $cpu_id != "0" ]]; then
+						# read the cpu id from the actual system
+						_cpu_id=$(cat $sysfs_path/channels/$rel_id/cpu)
+						# The current cpu id should be 0
+						if [[ $_cpu_id == '0' ]]; then
+							LogMsg "Verified the current cpu id is 0"
+						else
+							LogErr "Failed to verify the currenct cpu id. Expected 0 but found $_cpu_id"
+							FailedCount=$((FailedCount+1))
+						fi
+						# Change to random number
+						cpu_rdn=$(($RANDOM % $max_cpu))
+						echo $cpu_rdn > $sysfs_path/channels/$rel_id/cpu
+						# Read back new cpu id
+						_cpu_id=$(cat $sysfs_path/channels/$rel_id/cpu)
+						if [[ $_cpu_id == $cpu_rdn ]]; then
+							LogMsg "Verified the random number assigned to cpu successfully"
+						else
+							LogErr "Failed to verify random number assignment to the cpu id. Expected $cpu_rdn, found $_cpu_id"
+							FailedCount=$((FailedCount+1))
+						fi
+						# Change to the original cpu id
+						echo $cpu_id > $sysfs_path/channels/$rel_id/cpu
+						# Read back new cpu id
+						_cpu_id=$(cat $sysfs_path/channels/$rel_id/cpu)
+						if [[ $_cpu_id == $cpu_id ]]; then
+							LogMsg "Verified the original number assigned to cpu successfully"
+						else
+							LogErr "Failed to verify the original number assignment to the cpu id. Expected $cpu_id, found $_cpu_id"
+							FailedCount=$((FailedCount+1))
+						fi
+					fi
+					read -a line
+				done
+			else
+				LogErr "Failed. Supposed to read Sysfs path, but found ${line[@]}"
+				FailedCount=$((FailedCount+1))
+			fi
+			idx=$((idx+1))
+		else
+			LogErr "Failed. Supposed to read VMBUS ID line, but found ${line[@]}"
+			FailedCount=$((FailedCount+1))
+		fi
+		LogMsg ""
```

Comment:
Why do we need a blank log?

Reply:
There are many vmbus channels in VM. The number of vmbus channels and the number of cpu id are dependent on the VM size. Looping many times for all cpu id, it is good to have blank line for easy read. If you don't like it, I can remove it.

Reply:
Can we have a log message instead of blank line? like Verification complete?